docs/shared/concepts/sync-generators.md

```diff
@@ -0,0 +1,46 @@
+# Sync Generators
+
+In Nx 19.7, you can use sync generators to keep applicable files in sync with your project graph. These can be for global configuration files or scripts, or at the task level to ensure to that project level configuration files are in sync before a task is run.
+
+Sync Generator Examples:
+
+- Ensure code is formatted in a specific way before CI is run
+- Update a custom CI script with binning strategies based on the current project graph
+- Update TypeScript config files with project references based on the current project graph
+
+## Task Sync Generators
+
+Sync generators can be associated with a particular task. Nx will ensure that the sync generator is run before the task is executed (unless `sync.applyChanges` is set to `false` in `nx.json`). This is similar to the `dependsOn` property, but for generators instead of task dependencies.
```

Comment:
When `sync.applyChanges` is set to `false`, sync generators are still run before running tasks, but the user isn't prompted, and the changes aren't applied.

Note that this is only true when running in interactive environments. When running in CI or in a non-interactive environment, if the workspace is out of sync, an error will be logged, and the process will exit with code 1.

The `sync.applyChanges` allows you to set the default answer to the prompt that asks you whether you want to apply the changes. It doesn't allow you to opt-out of running the sync generators. It only allows you to opt-out of applying the identified changes needed to be made to sync the workspace.

To skip running sync generators when running tasks, you can run tasks with the `--skip-sync` flag or add specific sync generators to the `sync.disabledTaskSyncGenerators` array in `nx.json`.